### PR TITLE
hub home and side navigation

### PR DESCRIPTION
**Pull request by trickpattyFH20**

### CODE REVIEW

**Comment by zarembsky** [comment]
How about fancy one-liner in componentWillMount:
({title:window.document.title} = this.state);
instead of:
const { title } = this.state;
window.document.title = title;

**Comment by trickpattyFH20** [comment]
@zarembsky these should also be moved to the constructor since `componentWillMount` [is deprecated](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)

File: app/hub/Views/HomeView/HomeView.jsx [hunk 1]
```diff
@@ -43,7 +43,50 @@ class HomeView extends Component {
 	 */
 	render() {
 		const { title } = this.state;
-		return <div>{title}</div>;
+		return (
+			<div>
+				{title}
```
Comment (jsignanini) [inline app/hub/Views/HomeView/HomeView.jsx]: @trickpattyFH20 let's make sure this is an i18n-ized string

File: app/hub/Views/HomeView/HomeViewActions.js [hunk 2]
```diff
@@ -11,6 +11,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0
  */
 
-import TutorialTrustRestrictView from './TutorialTrustRestrictView';
-
-export default TutorialTrustRestrictView;
+export function toggleAnalytics(data) {
+	return {
+		type: 'TOGGLE_ANALYTICS',
```
Comment (jsignanini) [inline app/hub/Views/HomeView/HomeViewActions.js]: @trickpattyFH20 let'a make all action types `const` on their own constants file.

File: app/hub/Views/HomeView/HomeViewContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * @class Implement the Tutorial View Container for the Ghostery Hub
+ * @extends Component
+ * @memberof HubContainers
+ */
+class HomeViewContainer extends React.Component {
+	/**
+	 * Lifecycle Event
+	 */
+	componentWillMount() {
+		const title = 'Ghostery Hub - Home';
```
Comment (jsignanini) [inline app/hub/Views/HomeView/HomeViewContainer.jsx]: @trickpattyFH20 this should be i18n-ized.

File: app/hub/Views/TutorialView/TutorialView.jsx [hunk 4]
```diff
@@ -23,25 +23,34 @@ import TutorialNavigation from '../TutorialViews/TutorialNavigation';
...
-							<step.bodyComponent />
-						</div>
-					)}
-				/>
-			))}
-		</div>
+class TutorialView extends Component {
+	constructor(props) {
+		super(props);
+		this.props.history.push('/tutorial/1');
```
Comment (jsignanini) [inline app/hub/Views/TutorialView/TutorialView.jsx]: @trickpattyFH20 this should probably live in `componentDidMount()`?

File: app/hub/Views/TutorialView/TutorialViewActions.js [hunk 5]
```diff
@@ -19,3 +19,10 @@ export function initTutorialProps(data) {
 		data,
 	};
 }
+
+export function setTutorialNavigation(data) {
+	return {
+		type: 'SET_TUTORIAL_NAVIGATION',
```
Comment (jsignanini) [inline app/hub/Views/TutorialView/TutorialViewActions.js]: @trickpattyFH20 this should be a `const` imported from a constants file.

File: app/hub/Views/TutorialView/TutorialViewContainer.jsx [hunk 6]
```diff
@@ -38,6 +38,7 @@ class TutorialViewContainer extends React.Component {
 	componentWillMount() {
 		const title = '';
```
Comment (jsignanini) [inline app/hub/Views/TutorialView/TutorialViewContainer.jsx]: @trickpattyFH20 let's add the proper i18n-ized string here.

File: app/hub/Views/TutorialView/TutorialViewReducer.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+					},
+				}),
+			});
+		}
+
+		default: return state;
+	}
+}
+
+export default TutorialViewReducer;
```
Comment (jsignanini) [inline app/hub/Views/TutorialView/TutorialViewReducer.js]: @trickpattyFH20 all `case` strings here should be `const` imported from a constants file.

File: app/hub/Views/TutorialViews/TutorialAntiSuiteView/TutorialAntiSuiteView.jsx [hunk 8]
```diff
@@ -25,7 +25,7 @@ class TutorialAntiSuiteView extends Component {
 		super(props);
 
 		this.state = {
-			title: ''
+			title: 'Ghostery Hub - Tutorial Anti Suite'
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialAntiSuiteView/TutorialAntiSuiteView.jsx]: @trickpattyFH20 string should be i18n-ized.

File: app/hub/Views/TutorialViews/TutorialLayoutView/TutorialLayoutView.jsx [hunk 9]
```diff
@@ -16,16 +16,16 @@
...
  * @memberof HubComponents
  */
-class TutorailTrustRestrictView extends Component {
+class TutorialLayoutView extends Component {
 	constructor(props) {
 		super(props);
 
 		this.state = {
-			title: ''
+			title: 'Ghostery Hub - Tutorial Layout'
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialLayoutView/TutorialLayoutView.jsx]: @trickpattyFH20 string should be i18n-ized.

File: app/hub/Views/TutorialViews/TutorialBlockingView/TutorialBlockingViewContainer.jsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+		const { index } = this.props;
+		this.props.actions.setTutorialNavigation({
+			activeIndex: index,
+			hrefPrev: `/tutorial/${index - 1}`,
+			hrefNext: `/tutorial/${index + 1}`,
+			hrefDone: '/',
+			textPrev: t('hub_setup_nav_previous'),
+			textNext: t('hub_setup_nav_next'),
+			textDone: t('hub_setup_exit_flow'),
+		});
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialBlockingView/TutorialBlockingViewContainer.jsx]: @trickpattyFH20 should this be on `componentDidMount()` instead?

File: app/hub/Views/TutorialViews/TutorialLayoutView/TutorialLayoutViewContainer.jsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+		const { index } = this.props;
+		this.props.actions.setTutorialNavigation({
+			activeIndex: index,
+			hrefPrev: `/tutorial/${index - 1}`,
+			hrefNext: `/tutorial/${index + 1}`,
+			hrefDone: '/',
+			textPrev: t('hub_setup_nav_previous'),
+			textNext: t('hub_setup_nav_next'),
+			textDone: t('hub_setup_exit_flow'),
+		});
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialLayoutView/TutorialLayoutViewContainer.jsx]: @trickpattyFH20 should this be on `componentDidMount()` instead?

File: app/hub/Views/TutorialViews/TutorialNavigation/TutorialNavigationContainer.jsx [hunk 12]
```diff
@@ -12,26 +12,68 @@
...
+// 	render() {
+// 		const { totalSteps, tutorial } = this.props;
+// 		const { navigation } = tutorial;
+// 		const navigationProps = {
+// 			totalSteps,
+// 			...navigation,
+// 		};
+// 		return <SteppedNavigation {...navigationProps} />;
+// 	}
+// }
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialNavigation/TutorialNavigationContainer.jsx]: @trickpattyFH20 let's remove this leftover.

File: app/hub/Views/TutorialViews/TutorialTrackerListView/TutorialTrackerListViewContainer.jsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+		const { index } = this.props;
+		this.props.actions.setTutorialNavigation({
+			activeIndex: index,
+			hrefPrev: `/tutorial/${index - 1}`,
+			hrefNext: `/tutorial/${index + 1}`,
+			hrefDone: '/',
+			textPrev: t('hub_setup_nav_previous'),
+			textNext: t('hub_setup_nav_next'),
+			textDone: t('hub_setup_exit_flow'),
+		});
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialTrackerListView/TutorialTrackerListViewContainer.jsx]: @trickpattyFH20 should this be on `componentDidMount()` instead?

File: app/hub/Views/TutorialViews/TutorialTrustView/TutorialTrustView.jsx [hunk 14]
```diff
@@ -16,16 +16,16 @@
...
  * @memberof HubComponents
  */
-class TutorailSimpleDetailedView extends Component {
+class TutorialTrustView extends Component {
 	constructor(props) {
 		super(props);
 
 		this.state = {
-			title: ''
+			title: 'Ghostery Hub - Tutorial Trust'
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialTrustView/TutorialTrustView.jsx]: @trickpattyFH20 strings should be i18n-ized.

File: app/hub/Views/TutorialViews/TutorialTrustView/TutorialTrustViewContainer.jsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+		const { index } = this.props;
+		this.props.actions.setTutorialNavigation({
+			activeIndex: index,
+			hrefPrev: `/tutorial/${index - 1}`,
+			hrefNext: `/tutorial/${index + 1}`,
+			hrefDone: '/',
+			textPrev: t('hub_setup_nav_previous'),
+			textNext: t('hub_setup_nav_next'),
+			textDone: t('hub_setup_exit_flow'),
+		});
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialTrustView/TutorialTrustViewContainer.jsx]: @trickpattyFH20 should this be on `componentDidMount()` instead?

File: app/hub/Views/TutorialViews/TutorialVideoView/TutorialVideoViewContainer.jsx [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+		const { index } = this.props;
+		this.props.actions.setTutorialNavigation({
+			activeIndex: index,
+			hrefPrev: false,
+			hrefNext: `/tutorial/${index + 1}`,
+			hrefDone: '/',
+			textPrev: false,
+			textNext: t('hub_setup_nav_next'),
+			textDone: t('hub_setup_exit_flow'),
+		});
```
Comment (jsignanini) [inline app/hub/Views/TutorialViews/TutorialVideoView/TutorialVideoViewContainer.jsx]: @trickpattyFH20 should this be on `componentDidMount()` instead?

**Comment by trickpattyFH20** [comment]
`componentDidMount` [will be deprecated as of React 17](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)
the recommended migration is to move this to the constructor

**Comment by IAmThePan** [comment]
This was integrated into the `feature/side-navigation` pull request

**Comment by IAmThePan** [comment]
^^^ Pull Request #179

**Pull request closed without merging**